pkg/codegen/python/gen.go

```diff
@@ -1896,9 +1896,18 @@ func genPackageMetadata(
 	fmt.Fprintf(w, "from subprocess import check_call\n")
 	fmt.Fprintf(w, "\n\n")
 
-	// Create a constant for the version number to replace during build
-	fmt.Fprintf(w, "VERSION = \"0.0.0\"\n")
-	fmt.Fprintf(w, "PLUGIN_VERSION = \"0.0.0\"\n\n")
+	// If `pkg` supplies a version, use that. Otherwise, create a constant for
+	// the version number to replace during build.
+	pypiVersion := "0.0.0"
+	pkgVersion := "0.0.0"
+	if pkg.Version != nil {
+		pkgVersion = pkg.Version.String()
+	}
+	// TODO: puipyVersion is the pythonic version of pkgVersion. The code to
+	// transform it is in pulumi/pulumictl. Should that be extracted to
+	// somewhere like pulumi/pulumi/sdk/python.go?
```

Comment:
Yes, it does seem like we should extract that code and put it somewhere common. Not sure if `pulumi/pulumi/sdk` is the right place for that, but I also don't have a better suggestion offhand :) @komalali  @t0yv0 ?

Reply:
No strong opinions.. looks like pulumictl depends on pkg https://github.com/pulumi/pulumictl/blob/master/go.mod#L11 so if the least common ancestor is pkg we can put it there; alternatively factor into its own little dedicated GH repo and package. The barriers to that are pleasantly low. If it's really small then "a little dulication is better than a little dependency" applies here too so we can inline a copy.

Reply:
I just duplicated the code. 
Taking a closer look, most of the code dealt with the current git tree being dirty, and generating a semver version. Since we are provided a semver version already, I just help with the pythonic parts. It's not long. I located it in `pulumi/pkg/codegen/python/utilities.go`.